### ISSUE TITLE
parseArguments can return a non-array but the type is always an array.

### ISSUE THREAD

**Issue by jamie-pate** [int0h/typed-cli#15]
https://github.com/int0h/typed-cli/blob/2a85aa09e8d243672d31a67223d8b6fac6377d99/src/parser.ts#L86 calls `'normalizeArgs'(parsed._)`  which returns a string if there is one value, but the type of _ for `ResolveCliDeclaration<{_: Option<"string", false, false, string>}>` is `string[]`

e.g.
```TypeScript
const { _ } = cli({options: {}, _: option.string.label('path').description('Path to the file')})
```
produces `const _: string[] | undefined` but the value is actually a `string`

I think the easiest thing to do is to just stop calling `normalizeArgs` since any existing correct program expects that to be an array?

**Comment by int0h** [comment]
Hello! Sorry for the long response. I've checked what's going on. It was a bug in typing generation.

By design:
```js
const { _ } = cli({options: {}, _: option.string.label('path').description('Path to the file')})
```
`_` here should be either `string` or `undefined`. This code should be read as `define a program that takes 1 optional argument`

It might be usable for applications that always take only one arg. 

If the application takes more args `.array()` should be added and `_` will be `string[]`

I did the fix in `refresh` branch. And published `1.1.0-rc` version ini NPM. Please take a look if it solves your issue.

**Comment by jamie-pate** [comment]
LGTM
